controllers/storageclass/storageclass_controller.go

```diff
@@ -118,52 +118,69 @@ func (r *StorageClassWatcher) Reconcile(_ context.Context, request reconcile.Req
 		return result, err
 	}
 
-	flashSystemCluster, fscErr := r.getFlashSystemClusterByStorageClass(sc)
-	if fscErr == nil {
-		fscName := flashSystemCluster.GetName()
-		r.Log.Info("FlashSystemCluster found", "FlashSystemCluster", fscName)
-
-		// Check GetDeletionTimestamp to determine if the object is under deletion
-		if !sc.GetDeletionTimestamp().IsZero() {
-			r.Log.Info("object is terminated")
-			err = r.removeStorageClassFromConfigMap(*configMap, fscName, request.Name)
-			if err != nil {
-				return result, err
-			}
-		} else {
-			poolName, ok := sc.Parameters[util.CsiIBMBlockScPool]
-			if ok {
-				if err = r.addStorageClassToConfigMap(*configMap, fscName, request.Name, poolName); err != nil {
+	_, isTopology := sc.Parameters[util.TopologyStorageClassByMgmtId]
+	if isTopology {
+		r.Log.Info("StorageClass is a topology aware StorageClass", "isTopology", isTopology)
```

Comment:
no need to print the `isTopology` field, as it is clear we enter here only when `isTopology` is True.
Also, I think you can move this message to `getFlashSystemClusterByStorageClass` as we already check for `isTopology` there, and it's the only place we care about topology or non-topology storageclass.